packages/web-wallet/src/services/TokenRegistryService.ts

```diff
@@ -0,0 +1,293 @@
+    }
+
+    return tokenInfo;
+  }
+
+  /**
+   * Unregister a token
+   */
+  unregisterToken(tokenUid: string, network: string, genesisHash: string): void {
+    const tokens = tokenStorageService.loadTokens(network, genesisHash);
+    const filtered = tokens.filter(t => t.uid !== tokenUid);
+    const saved = tokenStorageService.saveTokens(network, genesisHash, filtered);
+
+    if (!saved) {
+      throw new Error('Failed to save token unregistration to browser storage. The token may reappear on page refresh.');
+    }
+
+    // Remove from cache
+    this.tokenCache.delete(tokenUid);
+  }
+
+  /**
+   * Validate configuration string format and checksum
+   */
+  validateConfigString(configString: string): ValidationResult {
+    try {
+      // Remove brackets if present
+      let cleaned = configString.trim();
+      if (cleaned.startsWith('[') && cleaned.endsWith(']')) {
+        cleaned = cleaned.slice(1, -1);
+      }
+
+      // Split into parts: name:symbol:uid:checksum
+      const parts = cleaned.split(':');
+      if (parts.length !== 4) {
+        return {
+          valid: false,
+          error: 'Invalid format. Expected: [name:symbol:uid:checksum]',
+        };
+      }
+
+      const [name, symbol, uid, providedChecksum] = parts;
+
+      // Validate each part
+      if (!name || name.length === 0) {
+        return { valid: false, error: 'Token name cannot be empty' };
+      }
+
+      if (!symbol || symbol.length === 0) {
+        return { valid: false, error: 'Token symbol cannot be empty' };
+      }
+
+      if (!uid || !/^[0-9a-f]{64}$/i.test(uid)) {
+        return {
+          valid: false,
+          error: 'Token UID must be 64 hexadecimal characters',
+        };
+      }
+
+      // Verify checksum
+      const partialConfig = `${name}:${symbol}:${uid}`;
+      const buffer = Buffer.from(partialConfig);
+      const calculatedChecksum = helpers.getChecksum(buffer).toString('hex');
+
+      if (calculatedChecksum !== providedChecksum.toLowerCase()) {
+        return {
+          valid: false,
+          error: 'Invalid checksum. The configuration string may be corrupted.',
+        };
+      }
+
+      return {
+        valid: true,
+        parsed: { uid, name, symbol },
+      };
+    } catch (error) {
+      return {
+        valid: false,
+        error: error instanceof Error ? error.message : 'Validation failed',
+      };
+    }
+  }
+
+  /**
+   * Fetch token metadata from wallet-lib
+   */
+  async fetchTokenMetadata(tokenUid: string, network: string): Promise<TokenMetadata> {
+    // Check cache first
+    const cached = this.tokenCache.get(tokenUid);
+    if (cached) {
+      return cached;
+    }
+
+    // Fetch from wallet-lib
+    try {
+      const tokens = await readOnlyWalletService.getTokens();
+      const tokenData = tokens.find((t: Record<string, unknown>) => t.uid === tokenUid) as Record<string, unknown> | undefined;
```

Comment:
Is iterating over all tokens the only way to get token data on the wallet-service facade?

Reply:
This has been heavily refactored, can you check this file again?